File: core/src/main/scala/diplomacy/variant/standard/rule/MilitaryBranch.scala
package diplomacy.variant.standard.rule

import diplomacy.board.{ MilitaryBranch => BaseMilitaryBranch, Name }

object MilitaryBranch {
  sealed trait MilitaryBranch extends BaseMilitaryBranch

  object Army extends MilitaryBranch {
    val name = Name("Army", "A")
  }
  object Fleet extends MilitaryBranch {
    val name = Name("Fleet", "F")
  }
}

File: lib/variant/standard/rule/military-branch.js
'use strict'

const { Name } = require('../../../board/package')

/**
 * Defines military branches
 * @namespace
 * @memberof variant.standard.rule
 */
const MilitaryBranch = {
  /**
   * Army
   * @type {!board.Name}
   */
  Army: new Name('Army', 'A'),
  /**
   * Fleet
   * @type {!board.Name}
   */
  Fleet: new Name('Fleet', 'F')
}

module.exports = MilitaryBranch
